src/super_gradients/training/processing/processing.py

```diff
@@ -344,6 +395,120 @@ def _get_padding_params(self, input_shape: Tuple[int, int]) -> PaddingCoordinate
         return _get_bottom_right_padding_coordinates(input_shape=input_shape, output_shape=self.output_shape)
 
 
+@register_processing()
+class DetectionAutoPadding(Processing):
+    def __init__(self, shape_multiple: Tuple[int, int], pad_value: int):
+        """
+        :param shape_multiple:  Tuple of (H, W) indicating the height and width multiples to which the input image dimensions will be padded.
+                                For instance, with a value of (32, 40), an input image of size (45, 67) will be padded to (64, 80).
+        :param pad_value:       Value to pad the image with.
+        """
+        self.shape_multiple = shape_multiple
+        self.pad_value = pad_value
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, DetectionPadToSizeMetadata]:
+        is_input_channel_first = image.ndim == 3 and image.shape[0] < image.shape[2]
```

Comment:
`image.shape[0] < image.shape[2]` this is quite hacky way to discriminate HWC / CHW layout.
I'm really starting to think we should have some intermediate structure holding this info:
```
@dataclass
class ImageContainer:
  data: Union[np.ndarray, torch.Tensor]
  layout: ImageLayout # HWC, CHW, BHWC, BCHW
  color_format: str # RGB / BGR / GRAY
```

Reply:
As discussed, I dont need it eventually [3874c86](https://github.com/Deci-AI/super-gradients/pull/1605/commits/3874c8672d1af4b11403307c69d367348ccad499)
ButI agree on the point for future improvement